pkgs/development/python-modules/pyzstd/default.nix

```diff
@@ -0,0 +1,45 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  pkgs,
+  pypaInstallHook,
+  setuptoolsBuildHook,
+}:
+buildPythonPackage rec {
+  pname = "pyzstd";
+  version = "0.16.2";
+
+  # There is a pyproject.toml, but we want to dynamically link zstd, which can
+  # only be done through a setup.py argument
+  pyproject = false;
+
+  src = fetchFromGitHub {
+    owner = "Rogdham";
+    repo = "pyzstd";
+    tag = version;
+    hash = "sha256-Az+0m1XUFxExBZK8bcjK54Zt2d5ZlAKRMZRdr7rPcss=";
+  };
+
+  nativeBuildInputs = [
+    pypaInstallHook
+    setuptoolsBuildHook
+  ];
+
+  buildInputs = [
+    # zstd from top-level scope, not python-modules scope
+    pkgs.zstd
```

Comment:
Would python3Packages.zstd actually be used here if we just took `zstd` as an argument directly instead of `pkgs`? Using `pkgs` should be avoided

Reply:
This was discussed previously: https://github.com/NixOS/nixpkgs/pull/365111#discussion_r1885009951

This is to do with how package scopes and `callPackage` works.

TL;DR `callPackage` gets it's auto-args from whatever _scope_ it is part of. In this case, the scope is `python3Packages`.

Reply:
Guessed as much

I hate nixpkgs